actions/testabilities/fixture_spvwallet_application.go

```diff
@@ -0,0 +1,179 @@
+package testabilities
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+
+	"github.com/bitcoin-sv/spv-wallet/config"
+	"github.com/bitcoin-sv/spv-wallet/engine"
+	"github.com/bitcoin-sv/spv-wallet/engine/datastore"
+	"github.com/bitcoin-sv/spv-wallet/engine/tester"
+	"github.com/bitcoin-sv/spv-wallet/engine/tester/fixtures"
+	"github.com/bitcoin-sv/spv-wallet/engine/utils"
+	"github.com/bitcoin-sv/spv-wallet/server"
+	"github.com/go-resty/resty/v2"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+)
+
+type ConfigOpts func(*config.AppConfig)
+
+type SPVWalletApplicationFixture interface {
+	StartedSPVWallet() (cleanup func())
+	StartedSPVWalletWithConfiguration(opts ...ConfigOpts) (cleanup func())
+
+	// HttpClient returns a new http client that can be used to make requests to the spv-wallet server.
+	// It is also failing tests if there are network or invalid request configuration errors,
+	// so the tests can focus on the server response.
+	HttpClient() SPVWalletHttpClientFixture
+
+	// NewTest creates a new test fixture based on the current one and the provided testing.TB
+	// This is useful if you want to start spv-wallet once and then run multiple t.Run with some calls against this one instance.
+	NewTest(t testing.TB) SPVWalletApplicationFixture
+}
+
+type SPVWalletHttpClientFixture interface {
+	// ForAnonymous returns a new http client that is configured without any authentication.
+	ForAnonymous() *resty.Client
+	// ForAdmin returns a new http client that is configured with the authentication with default admin xpub.
+	ForAdmin() *resty.Client
+	// ForUser returns a new http client that is configured with the authentication with the xpub of the sender user.
+	ForUser() *resty.Client
+}
+
+type appFixture struct {
+	config             *config.AppConfig
+	engine             engine.ClientInterface
+	t                  testing.TB
+	logger             zerolog.Logger
+	transport          testServer
+	dbConnectionString string
+	dbConnection       *sql.DB
+}
+
+func Given(t testing.TB) SPVWalletApplicationFixture {
+	f := &appFixture{
+		t:      t,
+		logger: tester.Logger(t),
+		config: getConfigForTests(),
+	}
+
+	f.initDbConnection()
+
+	return f
+}
+
+func (f *appFixture) NewTest(t testing.TB) SPVWalletApplicationFixture {
+	return newOf(*f, t)
+}
+
+func newOf(f appFixture, t testing.TB) *appFixture {
+	f.t = t
+	f.logger = tester.Logger(t)
+	return &f
+}
```

Comment:
The `newOf` is only used in `NewTest` function so maybe you could just move the code to the `NewTest` and skip the `newOf`?